src/scripts/main.js

```diff
@@ -1,3 +1,21 @@
 'use strict';
 
-// write your code here
+const populationElements = document.querySelectorAll('.population');
+const populations = Array.from(populationElements)
+  .map((el) => Number(el.textContent.replace(/,/g, '')))
+  .filter((num) => !isNaN(num));
+
+const totalPopulation = populations.reduce((sum, num) => sum + num, 0);
+const averagePopulation = totalPopulation / populations.length;
+
+function formatWithThousandsSeparator(number) {
+  return number.toLocaleString('en-US');
+}
+
+const averageElement = document.querySelector('.average-population');
+const totalElement = document.querySelector('.total-population');
```

Comment:
Ensure that the element with the class 'total-population' exists in the DOM before attempting to set its textContent. Otherwise, this line could throw an error if the element is not found.